### PR TITLE
Add a lens for viewing job pod information

### PR DESCRIPTION
**Pull request by Katharine**
This adds a low-priority lens that parses `podinfo.json` and gives you some information about the pod that ran your job. This appears unconditionally (assuming it's configured and that podinfo.json exists) - helpful insights as to why your job misbehaved will come soon to the status lens.

Some screenshots:

image
image
image
image
image

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
Is the time column from when the data was collected (a la `kubectl describe`) or from the timestamp on the event?

Can we visually distinguish between container details and other things? They're not really one-to-one with, say, events or the full YAML

**Comment by Katharine** [comment]
> Is the time column from when the data was collected (a la kubectl describe) or from the timestamp on the event?

Assuming you're referring to the events, it's counting from when the event occurred to the time you view the page (so the same as if you were to run `kubectl describe` at the time you loaded the page).

> Can we visually distinguish between container details and other things? They're not really one-to-one with, say, events or the full YAML

I agree, but actually designing that is tricky. I'm thinking maybe have the last tab be "containers" and clicking it gets you a dropdown of containers to pick from? it's not particularly elegant and would be something of a pain to implement.

What are your thoughts on iteratively improving this?

**Comment by matthyx** [comment]
I like it this way... it's already much better than nothing.
I would suggest to rollout and iterate later...

File: prow/spyglass/lenses/podinfo/podinfo.go [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+	"k8s.io/test-infra/prow/entrypoint"
+	"sigs.k8s.io/yaml"
+
+	"k8s.io/test-infra/prow/spyglass/lenses"
+)
+
+const (
+	name     = "podinfo"
+	title    = "Job Pod Info"
+	priority = 20
```
[the remark this answers is not in the document]
Comment (Katharine) [reply prow/spyglass/lenses/podinfo/podinfo.go]: The current highest priority is 10, so this puts it at the bottom of the page with room to add things in between.

**Pull request merged**